### PR TITLE
[feat-4864]: Backend support for project deletion

### PR DESCRIPTION
**Pull request by keon94**
### Summary
Project deletion support

### Does this close any open issues?
Child of #4864 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/helpers/pluginhelper/services/db_helper.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/apache/incubator-devlake/core/plugin"
 )
 
-type BlueprintManager struct {
+type DatabaseHelper struct {
```
Comment (keon94) [inline backend/helpers/pluginhelper/services/db_helper.go:29]: more appropriate (and general name)

File: backend/server/services/project.go [hunk 2]
```diff
@@ -254,6 +247,65 @@ func PatchProject(name string, body map[string]interface{}) (*models.ApiOutputPr
...
+	}
+	err = tx.Delete(&crossdomain.ProjectPrMetric{}, dal.Where("project_name = ?", name))
+	if err != nil {
+		return err
+	}
+	err = tx.Delete(&crossdomain.ProjectIssueMetric{}, dal.Where("project_name = ?", name))
+	if err != nil {
+		return err
+	}
+	err = tx.Commit()
```
Comment (klesh) [inline backend/server/services/project.go:290]: Let us delete the related `blueprint` as well. It is different than deleting related scopes, we are sure the bp is useless when the corresponding project is gone.

**Review by klesh — COMMENTED** [review]
LGTM overall, only one minor problem to be fixed or discussed

File: backend/server/services/project.go (hunk 2 above)
Comment (keon94) [reply backend/server/services/project.go:290]: Done

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**